addons/pos_restaurant/static/tests/tours/utils/devices_synchronization.js

```diff
@@ -0,0 +1,130 @@
+/* global posmodel */
+
+const getData = ({ lineProductName, productName, partnerName } = {}) => {
+    const order = posmodel.models["pos.order"].find((o) => o.pos_reference === "device_sync");
+
+    let partner = null;
+    if (partnerName) {
+        partner = posmodel.models["res.partner"].find((p) => p.name === partnerName);
+    }
+
+    let line = null;
+    if (lineProductName) {
+        line = order.lines.find((l) => l.product_id.display_name === lineProductName);
+    }
+
+    let product = null;
+    if (productName) {
+        product = posmodel.models["product.product"].find((p) => p.display_name === productName);
+    }
+
+    return { order, line, product, partner };
+};
+
+const notify = async () => {
+    const orm = posmodel.env.services.orm;
+    await orm.call("pos.config", "notify_synchronisation", [
+        posmodel.config.id,
+        posmodel.session.id,
+        999,
+    ]);
+};
+
+const getLineData = (product, order, quantity) => ({
+    name: product.display_name,
+    order_id: order.id,
+    product_id: product.id,
+    price_unit: product.lst_price,
+    price_subtotal: product.lst_price * quantity,
+    price_subtotal_incl: product.lst_price * quantity,
+    discount: 0,
+    qty: quantity,
+});
+
+// In the point-of-sale code, we consider that synchronization is necessary
+// when the write_date of the local order is smaller than that of the server.
+// To prevent the PoS from ignoring our synchronization, we have to wait
+// for it to be taken into account.
+// Tours are to fast and sometimes the server write_date is the same.
+const writeOnOrder = (order, data) =>
+    new Promise((resolve) => {
+        const orderSeconds = new Date(order.write_date).getSeconds();
+        let serverSeconds = new Date().getSeconds();
+
+        while (orderSeconds === serverSeconds) {
+            serverSeconds = new Date().getSeconds();
+            if (orderSeconds !== serverSeconds) {
+                break;
+            }
+        }
+
+        (async () => {
+            const orm = posmodel.env.services.orm;
+            await orm.write("pos.order", [order.id], data);
+            resolve();
+        })();
+    });
```

Comment:
```suggestion
const writeOnOrder = async (order, data) => {
    await new Promise(resolve => setTimeout(resolve, 1000))
    const orm = posmodel.env.services.orm;
    await orm.write("pos.order", [order.id], data);
};
```
That will wait for 100ms before requesting a write.